app/containers/CreateTag/Form.jsx

```diff
@@ -104,55 +96,41 @@ let FormClone = reduxForm({
   onSubmitFail: (errors) => scrollToErrorField(errors),
 })(Form);
 
-FormClone = connect(
-  (state, { communities, communityId, isEditTagForm, editTagData }) => {
-    // map state to props for editTag form
-    if (isEditTagForm) {
-      const { communityId, tagId } = editTagData;
-      const existingTags = selectExistingTags()(state);
-      const communityTags = Array.isArray(existingTags)
-        ? existingTags
-        : existingTags[communityId];
-      const selectedTag = communityTags.find((tag) => tag.id === tagId);
-
-      const selectedCommunity = communities.find(
-        (comm) => comm.id === communityId,
-      );
+FormClone = connect((state, { communities, communityId, isEditTagForm, editTagData }) => {
+  const existingTags = selectExistingTags()(state);
+  // map state to props for editTag form
+  if (isEditTagForm) {
+    const { communityId, tagId } = editTagData;
+    const communityTags = Array.isArray(existingTags) ? existingTags : existingTags[communityId];
+    const selectedTag = communityTags.find((tag) => tag.id === tagId);
 
-      return {
-        valueHasNotBeInListValidate: communityTags
-          .filter((tag) => tag.id !== tagId)
-          .map((x) => x.name?.toLowerCase())
-          .concat(
-            (state?.toJS()?.tags?.suggestedTags ?? []).map((x) =>
-              x.name?.toLowerCase(),
-            ),
-          ),
-        initialValues: {
-          [FORM_COMMUNITY]: selectedCommunity,
-          [NAME_FIELD]: selectedTag?.name,
-          [DESCRIPTION_FIELD]: selectedTag?.description,
-        },
-        enableReinitialize: true,
-      };
-    }
+    const selectedCommunity = communities.find((comm) => comm.id === communityId);
 
-    // map state to props for createTag form
     return {
-      valueHasNotBeInListValidate: (
-        state?.toJS()?.form?.[FORM_NAME]?.values?.[FORM_COMMUNITY]?.tags ?? []
-      )
+      valueHasNotBeInListValidate: communityTags
+        .filter((tag) => tag.id !== tagId)
         .map((x) => x.name?.toLowerCase())
-        .concat(
-          (state?.toJS()?.tags?.suggestedTags ?? []).map((x) =>
-            x.name?.toLowerCase(),
-          ),
-        ),
+        .concat((state?.toJS()?.tags?.suggestedTags ?? []).map((x) => x.name?.toLowerCase())),
       initialValues: {
-        [FORM_COMMUNITY]: getFollowedCommunities(communities, [communityId])[0],
+        [FORM_COMMUNITY]: selectedCommunity,
+        [NAME_FIELD]: selectedTag?.name,
+        [DESCRIPTION_FIELD]: selectedTag?.description,
       },
+      enableReinitialize: true,
     };
-  },
-)(FormClone);
+  }
+
+  // map state to props for createTag form
+  return {
+    valueHasNotBeInListValidate: (
+      existingTags[state?.toJS()?.form?.[FORM_NAME]?.values?.[FORM_COMMUNITY]?.id] ?? []
+    )
+      .map((x) => x.name?.toLowerCase())
+      .concat((state?.toJS()?.tags?.suggestedTags ?? []).map((x) => x.name?.toLowerCase())),
```

Comment:
do not use such an abbreviation "x". Replace it elsewhere

Reply:
old code, but done